x/staking/src/types/domain.rs

```diff
@@ -328,3 +387,82 @@ impl Validator {
         key
     }
 }
+
+impl TryFrom<ValidatorRaw> for Validator {
+    type Error = Error;
+    fn try_from(value: ValidatorRaw) -> Result<Self, Self::Error> {
+        let status = value.status();
+        Ok(Self {
+            operator_address: ValAddress::from_bech32(&value.operator_address)
+                .map_err(|e| Error::DecodeAddress(e.to_string()))?,
+            delegator_shares: Decimal256::from_str(&value.delegator_shares)
+                .map_err(|e| Error::DecodeGeneral(e.to_string()))?,
+            description: value
+                .description
+                .expect("Value should exists. It's the proto3 rule to have Option<T> instead of T"),
```

Comment:
Is this guaranteed not to panic? Perhaps return an error?

Reply:
Small comment. proto3 doesn't have rule for all fields to be optional if you meant this. It makes all fields optional by default

Reply:
The protobuf conversion says that messages aka serialized structs are serialized to `Option<T>`. Please, look at [this section](https://docs.rs/prost/latest/prost/#field-modifiers). 
Since I wrote serialization from original type to raw version I expect that the protobuf cannot miss field of original type. It's like `unwrap_or_corrupt`. 

If you disagree with this thoughts I'll update the code.

Reply:
Only the code calling this method knows whether to `expect` these fields to be present e.g. if the calling code is reading serialized bytes from state then that code can `expect` the fields to be present. If however the calling code is deserializing a Transaction then that code would need handle the error case (you can't guarantee that the submitted transaction was created using your code).
TLDR: I don't think you can safely unwrap here since you don't have enough context.